File: week_6/mario.py
from cs50 import get_int

# question
while True:
    height = get_int("Height: ")
    if height in [1, 2, 3, 4, 5, 6, 7, 8]:
        break
  
# number of lines
for i in range(height):
    
    # number of spaces before #    
    for j in range(height - i - 1):
        print(" ", end="")
    
    # number of #        
    for h in range(i + 1):
        print("#", end="")
    
    # space        
    print("  ", end="")
    
    # number of #2    
    for g in range(i + 1):
        print("#", end="")
    
    # new line
    print("")

File: week_1/mario.c
#include <stdio.h>
#include <cs50.h>

// print mario pyramid, i.e. for height 4 print
//    # #
//   ## ##
//  ### ###
// #### ####


int main(void)
{
    // user input
    int n;
    do
    {
        n = get_int("height: ");
    }
    while (n < 1 || n > 8); // only accept range 1-8 (inclusive)

    // print lines
    for (int i = 0; i < n; i++)
    {
        // print spaces before first #
        for (int h = 0; h < n - i - 1; h++)
        {
            printf(" ");
        }

        // print left side of line
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // print gap
        printf("  ");

        // print right side of line
        for (int g = 0; g < i + 1; g++)
        {
            printf("#");
        }
        
        printf("\n");
    }
}